File: src/main/java/me/jko/javatokotlin/service/UserService.java
package me.jko.javatokotlin.service;

import me.jko.javatokotlin.entity.User;
import me.jko.javatokotlin.repository.UserRepository;
import org.springframework.stereotype.Service;

import java.util.Optional;

@Service
public class UserService {

    private final UserRepository userRepository;

    public UserService(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    public User createUser(String name, String email) {
        User user = new User();
        user.setName(name);
        user.setEmail(email);

        return userRepository.save(user);
    }

    public Optional<User> getUserById(Long id) {
        return userRepository.findById(id);
    }

    public User updateUsername(Long id, String updateName) {
        User user = userRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("User not found with id: " + id));

        user.setName(updateName);

        return userRepository.save(user);
    }

    public void deleteUser(Long id) {
        userRepository.deleteById(id);
    }
}

File: src/main/java/me/jko/javatokotlin/service/UserService.kt
package me.jko.javatokotlin.service

import me.jko.javatokotlin.entity.User
import me.jko.javatokotlin.repository.UserRepository
import org.springframework.stereotype.Service
import java.util.*

@Service
class UserService(private val userRepository: UserRepository) {

    fun createUser(name: String, email: String): User {
        val user = User()
        user.name = name
        user.email = email

        return userRepository.save(user)
    }

    fun getUserById(id: Long): Optional<User> {
        return userRepository.findById(id)
    }

    fun updateUsername(id: Long, updateName: String): User {
        val user = userRepository.findById(id)
            .orElseThrow {
                RuntimeException(
                    "User not found with id: $id"
                )
            }

        user.name = updateName

        return userRepository.save(user)
    }

    fun deleteUser(id: Long) {
        userRepository.deleteById(id)
    }
}
